node/taskcommand.ts

```diff
@@ -33,12 +34,16 @@ export class TaskCommand {
                     var val = this.properties[key];
                     if (val) {
                         cmdStr += key + '=' + val + ';';
+                        if(key === 'secret' && val === 'true') {
+                            containsSecret = true;
+                        }
                     }                    
                 }
             }
         }
 
-        cmdStr += ']' + this.message;
+        cmdStr += ']';
+        cmdStr += containsSecret? '********' : this.message;
```

Comment:
Actually setVariable() is broken if secret is true. If we don't mask in the command, the value is written without being masked. If I mask it, the command actually does not get the value i.e. the variable is not set for future use during the build. In my case the fix works since I don't use the variable again but the fix is not correct.